test/functional/tag_controller_test.rb

```diff
@@ -143,41 +159,39 @@ def setup
   end
 
   test 'tag show range' do
-    get :show, id: tags(:spectrometer).name,
+    get :show, params: { id: tags(:spectrometer).name,
                start: (Time.now - 1.day).strftime('%d-%m-%Y'),
-               end: Time.now.strftime('%d-%m-%Y')
+               end: Time.now.strftime('%d-%m-%Y') }
 
     assert :success
     assert_not_nil :tags
   end
 
   test 'tag show JSON' do
-    get :show, id: tags(:spectrometer).name, format: 'json'
+    get :show, params: { id: tags(:spectrometer).name, format: 'json' }
 
     assert :success
     assert_not_nil :tags
-
     json = ActiveSupport::JSON.decode(@response.body)
```

Comment:
Hi @jywarren, when I print this json variable. I comes something like this:
`{"node"=>{"nid"=>1, "vid"=>0, "type"=>"note", "language"=>"", "title"=>"Canon A1200 IR conversion at PLOTS Barnraising at LUMCON", "uid"=>2, "status"=>1, "created"=>1528171393, "changed"=>[], "comment"=>0, "promote"=>0, "moderate"=>0, "sticky"=>0, "tnid"=>0, "translate"=>0, "cached_likes"=>0, "comments_count"=>0, "drupal_node_revisions_count"=>0, "main_image_id"=>nil, "slug"=>"jeff-06-05-2018-canon-a1200-ir-conversion-at-plots-barnraising-at-lumcon", "legacy_views"=>0, "views"=>0}, "path"=>"https://test.host/notes/jeff/06-05-2018/canon-a1200-ir-conversion-at-plots-barnraising-at-lumcon", "preview"=>"\nHere is a #hashtag1, and another #hashtag2!\n\nHere is another #hashtag3.\n\n", "tags"=>["test", "awesome", "spectrometer", "activity:spectrometer"]}`.
Here you can see that "nid" hash is nested inside "node" hash
so  shouldn't it be ["node"]["nid"] instead of only ["nid"]

Reply:
I'll start by looking to see if this test was written concurrently with a particular JS function.

Reply:
Ooh, ok good leads here: https://github.com/publiclab/plots2/blame/master/test/functional/tag_controller_test.rb#L154

Reply:
The test was originally written here: https://github.com/publiclab/plots2/pull/1147/files -- it looks like it ought to return an array of nodes, so I don't know why `node` would be needed. You can see sample output here:

http://publiclab.org/tag/particle-imaging.json

which indeed has `nid` and not sub-objects of `node`. Let me check your code for how the JSON was encoded...

Reply:
Yeah according the json file you gave as the sample output it should be `['nid']` instead of `['node']['nid']`. Let me check why the output of json is coming like this in tests.

Reply:
So I think this section is unaffected - 

https://github.com/Souravirus/plots2/blob/0231d62f452a27baf90bb2b53484ed8a2a1745e3/app/controllers/tag_controller.rb#L136-L142

And maybe you should look at the `as_json` function to ensure it's still returning `node` objects, or maybe it's now returning `{node: {...}}` objects, like, wrapped in an extra object?